### ISSUE TITLE
the thumt transformer model no implement early stopping to prevent overfitting?

### ISSUE THREAD

**Issue by zxqchat** [thumt/THUMT#47]
the early stopping to prevent overfitting is very useful to train a NMT model,but I didn't found it in thumt transformer.

**Comment by Glaceon31** [comment]
Early stopping can be achieved in two ways in THUMT:
1. When `--validation <your validation source file> --references <your validation reference files>` is set in training, THUMT will evaluate on the validation set periodically. The checkpoints that achieves highest BLEU scores on the validation set will be saved in the train/eval folder. (You can halt or resume the training whenever you want.)
2. Setting `keep_checkpoint_max` in parameters to reserve as many checkpoints as you want.
